Menus/MainMenuGUI.cpp

```diff
@@ -635,8 +635,42 @@ void MainMenuGUI::Update()
         return;
 
     // If esc pressed, show quit dialog if applicable
-    if (g_UInputMan.KeyPressed(KEY_ESC))
-        QuitLogic();
+	if (g_UInputMan.KeyPressed(KEY_ESC)) {
+		if (m_MenuScreen == OPTIONSSCREEN) {
+			HideAllScreens();
+			m_MainMenuButtons[BACKTOMAIN]->SetVisible(false);
+
+			g_SettingsMan.SetFlashOnBrainDamage(m_aOptionsCheckbox[FLASHONBRAINDAMAGE]->GetCheck());
+			g_SettingsMan.SetBlipOnRevealUnseen(m_aOptionsCheckbox[BLIPONREVEALUNSEEN]->GetCheck());
+			g_SettingsMan.SetShowForeignItems(m_aOptionsCheckbox[SHOWFOREIGNITEMS]->GetCheck());
+			g_SettingsMan.SetShowToolTips(m_aOptionsCheckbox[SHOWTOOLTIPS]->GetCheck());
+			g_SettingsMan.SetPreciseCollisions(m_aOptionsCheckbox[PRECISECOLLISIONS]->GetCheck());
+			g_SettingsMan.UpdateSettingsFile();
+
+			m_MenuScreen = MAINSCREEN;
```

Comment:
There's a lot of duplicated code you could easily unduplicate with a bit of shifting around (hiding all screens, setting m_MenuScreen, etc.)
If you don't wanna make an agglomerated if check for everything that doesn't make it quit, you could consider moving this into a private method and returning after quit, and doing the duplicated stuff outside of the if check, i.e.
```
void MainMenuGUI::DoThing() {
	if (thing) {
		...
	} else if (otherThing) {
		...
	} else {
		QuitLogic();
		return;
	}
	HideAllScreens();
	m_MenuScreen = MAINSCREEN;
	...
}
```

This might be a little less easy to follow than an agglomerated if check or something, do what you feel works well.

Reply:
No problem, I'll do your first suggestion. I'm not looking to add methods or anything pretty like that.